Modules/+Experiments/+AutoExperiment/@SpecSlowScan/analyze.m

```diff
@@ -672,13 +672,19 @@ function save_state()
                 analysis.sites(site_index,i-1).fit = fit_result;
                 analysis.sites(site_index,i-1).amplitudes = fitcoeffs(1:nn);
                 analysis.sites(site_index,i-1).locations = fitcoeffs(nn+1:2*nn);
+                % Add widths to structure
                 if strcmpi(FitType,'gauss')
                     analysis.sites(site_index,i-1).widths = fitcoeffs(2*nn+1:3*nn)*2*sqrt(2*log(2));
                 else
                     analysis.sites(site_index,i-1).widths = fitcoeffs(2*nn+1:3*nn);
                 end
+                % Add etas to structure
                 if strcmpi(FitType,'voigt')
                     analysis.sites(site_index,i-1).etas = fitcoeffs(3*nn+2:4*nn+1);
+                elseif strcmpi(FitType,'gauss')
+                    analysis.sites(site_index,i-1).etas = zeros(1,nn);
+                elseif strcmpi(FitType,'lorentz')
+                    analysis.sites(site_index,i-1).etas = ones(1,nn);
                 end
                 analysis.sites(site_index,i-1).background = fitcoeffs(3*nn+1);
```

Comment:
Just for sake of future errors, let's move this (and the above) width logic into a switch statement. I hate how it is so un-modular, but until we move fitting to be more class-based, here we are. so something like:
```matlab
switch lower(FitType)
    case 'gauss'
        ....width code...
        ....eta code...
    case ...
    otherwise
        error('Unsupported')
end
```